cimi-resources/test/com/sixsq/slipstream/ssclj/resources/external_object_lifecycle_test_utils.clj

```diff
@@ -108,361 +118,324 @@
+                                                   :body)
+                  view-rule {:principal username-view
+                             :type      "USER"
+                             :right     "VIEW"}
+
+                  updated-rules (vec (conj (:rules acl) view-rule))
+
+                  updated-eo (-> current-eo
+                                 (assoc-in [:acl :rules] updated-rules)
+                                 (assoc :name "NEW_VALUE_OK"
+                                        :state "BAD_VALUE_IGNORED"))]
+
+              (-> session
+                  (request abs-uri
+                           :request-method :put
+                           :body (json/write-str updated-eo))
                   (ltu/body->edn)
-                  (ltu/is-status 201)
-                  (ltu/location))
+                  (ltu/is-status 200)))
+
+            ;; retrieve by another authorized user MUST NOW SUCCEED
+            ;; verify also that name can be updated, but not state
+            (let [updated (-> session-user-view
+                              (request abs-uri)
+                              (ltu/body->edn)
+                              (ltu/is-status 200)
+                              :response
+                              :body)]
+
+              (is (= "NEW_VALUE_OK" (:name updated)))
+              (is (not= "BAD_VALUE_IGNORED" (:state updated))))
+
+            ;; anonymous query fails
+            (-> session-anon
+                (request base-uri)
+                (ltu/body->edn)
+                (ltu/is-status 403))
+
+            ;; owner query succeeds
+            (let [entry (-> session
+                            (request base-uri)
+                            (ltu/body->edn)
+                            (ltu/is-status 200)
+                            (ltu/is-resource-uri eo/collection-uri)
+                            (ltu/is-count 1)
+                            (ltu/entries eo/resource-tag)
+                            first)
+                  id (:id entry)
+                  abs-uri (str p/service-context id)]
+
+              (is (= id uri))
+
+              (let [upload-op (-> session
+                                  (request abs-uri)
+                                  (ltu/body->edn)
+                                  (ltu/is-operation-present "upload")
+                                  (ltu/is-operation-present "delete")
+                                  (ltu/is-operation-present "edit")
+                                  (ltu/is-operation-absent "ready")
+                                  (ltu/is-operation-absent "download")
+                                  (ltu/is-status 200)
+                                  (ltu/get-op "upload"))
+
+                    abs-upload-uri (str p/service-context (u/de-camelcase upload-op))]
+
+                ;; triggering the upload url with anonymous, authorized or unauthorized viewer should fail
```

Comment:
The 3 requests could be grouped in a `doseq`

Reply:
Done.